environs/instances/instance.go

```diff
@@ -14,6 +14,10 @@ type Instance interface {
 	// Id returns a provider-generated identifier for the Instance.
 	Id() instance.Id
 
+	// DisplayName returns a provider-generated, user-friendly
+	// instance name for display purposes.
+	DisplayName() instance.DisplayName
```

Comment:
I've added `DisplayName` to the instance interface. Although that means that every provider needs to implement it, we get the benefit of type safety. Let me know if this is overkill.

Reply:
Yeah, not needed. DisplayName is just a string

Reply:
Just to clarify. Keep the method, but use string as the type?